internal/stores/hostdb.go

```diff
@@ -133,15 +135,24 @@ func (db *SQLStore) Host(hostKey consensus.PublicKey) (hostdb.Host, error) {
 
 // Hosts returns up to max hosts that have not been interacted with since
 // the specified time.
-func (db *SQLStore) Hosts(notSince time.Time, max int) ([]hostdb.Host, error) {
-	// Filter all hosts for the ones that have not been updated since a
-	// given time.
+func (db *SQLStore) Hosts(offset, limit int, whitelist []string) ([]hostdb.Host, error) {
+	if offset < 0 {
+		return nil, ErrNegativeOffset
+	}
+
 	var fullHosts []dbHost
-	err := db.db.Table("hosts").
-		Joins("LEFT JOIN host_interactions ON host_interactions.db_host_id = hosts.ID").
-		Group("Public_Key").
-		Having("IFNULL(MAX(Timestamp), 0) < ?", notSince.UTC()). // use UTC since we stored timestamps in UTC
-		Limit(max).
+	tx := db.db.Model(&dbHost{}).
+		Joins("LEFT JOIN announcements ann ON hosts.id = ann.db_host_id")
+
+	if len(whitelist) > 0 {
+		tx = tx.Where("ann.net_address IN (?)", whitelist)
+	}
+
+	err := tx.
+		Group("hosts.id").
+		Having("MAX(IFNULL(ann.timestamp,1))").
```

Comment:
2 questions.

1. Shouldn't `Having` always contain a condition that evaluates to true or false?
2. Why are we still looking at the timestamp if we no longer use `notSince` ? 

Without `notSince` it seems like we would just need to `Find` all hosts, populate their interactions and announcements and apply the offset and limit?